public/app/features/explore/state/query.ts

```diff
@@ -593,13 +574,19 @@ export const runQueries = (
         }
       } else {
         for (const type of supplementaryQueryTypes) {
-          // We always prepare provider, even is supplementary query is disabled because when the user
-          // enables the query, we need to load the data, so we need the provider
-          if (hasSupplementaryQuerySupport(datasourceInstance, type)) {
-            const dataProvider = datasourceInstance.getDataProvider(type, {
+          const dataProvider = getSupplementaryQueryProvider(
+            datasourceInstance,
+            type,
+            {
               ...transaction.request,
               requestId: `${transaction.request.requestId}_${snakeCase(type)}`,
-            });
+            },
+            newQuerySource
+          );
+
+          // We always prepare provider, even is supplementary query is disabled because when the user
+          // enables the query, we need to load the data, so we need the provider
```

Comment:
Is this the right place for the comment? Should't it be above `const dataProvider =...`

Reply:
Fixed in 7da84bd360ccdf3eec5605ed4af27a2c03a73621